CODE_STRUCTURE.md

```diff
@@ -0,0 +1,176 @@
+1. Typealiases for the Reducer, Store and a ViewStore.
+2. State
+3. Action
+4. Environment - always present even when no dependencies are defined = never use Void
+5. Reducer holding private hashable Ids, always use `static let default` reducer. Try to avoid `default: return .none` for the actions, we want the switch exhaustivenes check. 
+6. Store
+7. Viewstore
+8. Placeholders
+
+Any extension needed is defined under the relevant section except placeholders, those are separated at the very end of the file. Example: 
+
+```swift
+// MARK: - State
+
+struct <feature name>State: Equatable { }
+
+extension <feature name>State {
+    // anything but placeholder
+}
+
+// MARK: - Action
+// ...
+```
+
+##### Structure of Store & View files
+
+Use the appropriate name for a feature you want to build. The <feature name> must be bare bone of any TCA definition including the folder encapsulating the swift file(s). Example:
+
+```swift
+// Project structure
+// Folder <feature name> woth following files:
+
+// <feature name>Store.swift
+struct <feature name>State: Equatable { }
+enum <feature name>Action: Equatable { }
+struct <feature name>Environment
+
+// <feature name>View.swift
+struct <feature name>: View {
+    let store: <feature name>Store
+    
+    var body: some View {
+        WithViewStore(store) { viewStore in EmptyView() }
+    }
+}
+
+struct <feature name>_Previews: PreviewProvider {
+    static var previews: some View {
+    <feature name>(store: .placeholder)
+    }
+}
+```
+
+Do not omit Previews!
+
+##### Navigation
+Navigation is done using routes. Such mechanism consists of Route definition + state and Route action to update the state.
+```swift
+struct <feature>State
+enum Route {
+    case <route name>
+}
+var route: Route?
+}
+
+enum <feature>Action {
+    case updateRoute(<feature>State.Route?)
+}
+```
+
+##### XCode Template
+We created a TCA template that automatically prepares the TCA files and code with the rules defined in this document. You can download it [here].
+
+In order to put it to the XCode, do the following:
+1. open Finder
+2. press cmd + shift + G
+3. paste `/Applications/Xcode.app/Contents/Developer/Library/Xcode/Templates/File Templates`
+4. copy paste the TCA.xctemplate to the `MultiPlatform/Source/`
+
+The template is customizable and consists of 3 sections:
+
+1. TCA Store text field: that is essentially the <feature name>
+2. Selector of either
+a. Empty (= TCA file with no navigation included)
+b. Navigation (= TCA file with the navigation)
+3. Checkbox which allows you to create also a View file if requested.
+
+## Project Structure
+The project is specifically designed to be structured as follows:
+```
+Models
+    <model>.swift
+Wrappers
+    Wrapped<dependency name>.swift
+Utils
+    <util>.swift
+Dependencies
+    <dependency name>.swift
+Features
```

Comment:
I think they used SPM (modules) mostly to prove the composability and independency of the architecture. I don't think we need to do the same approach in our case. I don't see any clear added value because our dependencies are specific for the wallet and the sdk. The way isowords is done is also not necessarily making those modules re-usable. Most of it is not as they are specific modules for the game. We would end up with a tree of dependencies anyway. Re-using 1 module from isowords in some cases means import several modules (the one is dependent of many other and those might be dependent on other as well, clearly noticeable at one of the pointfree videos). 

With a few exceptions like User Defaults module in isowords, an example of a module that can be separated and in the SPM but it's so lightweight that anybody should probably implement its own version, with the needed structure and set of methods (closures).

Reply:
I was not necessarily referring to isowords, because that app was hyper modularized to make a point. I'm referring to the "feature modules" they proposed on another of the sample apps that they stated it paid dividends on build times which I can't seem able to find.